src/chess_teacher_stockfish/main.py

```diff
@@ -79,11 +79,17 @@ class AnalyzeMoveRequest(EvaluatePositionRequest):
     move: str
 
 
+class EvaluationResult(CamelCaseAliasBaseModel):
+    """Class representing the evaluation result from stockfish"""
+
+    value: Union[str, int]
+
+
 class EvaluatePositionResponse(CamelCaseAliasBaseModel):
     """Response object containing the Stockfish evaluation of the current position,
     the win/draw/loss statistics and the top three moves"""
 
-    evaluation: dict[str, str | int]
+    evaluation: dict[str, EvaluationResult]
```

Comment:
That is not really what was meant with the issue. `evaluation` is still a dictionary, but can be represented with a class, just like `TopMove`. Look at a sample JSON response and consider how it can be represented using a class:

```json
"evaluation": {
  "type": "cp",
  "value": 33
},
```